src/field/scalar.rs

```diff
@@ -260,6 +265,71 @@ impl Scalar {
 
         result
     }
+
+    /// Attempt to construct a `Scalar` from a canonical byte representation.
+    ///
+    /// # Return
+    ///
+    /// - `Some(s)`, where `s` is the `Scalar` corresponding to `bytes`,
+    ///   if `bytes` is a canonical byte representation;
+    /// - `None` if `bytes` is not a canonical byte representation.
+    pub fn from_canonical_bytes(bytes: [u8; 57]) -> Option<Scalar> {
+        // Check that the 10 high bits are not set
+        if bytes[56] != 0u8 || (bytes[55] >> 6) != 0u8 {
+            return None;
+        }
+        let bytes: [u8; 56] = std::array::from_fn(|i| bytes[i]);
```

Comment:
First time seeing `std::array::from_fn`, learn something new each day :)

Reply:
I just learned about it myself, while searching for a less annoying way to get a subarray from a array :grin: 

BTW --- RFC 8032 specifies that scalars should be serialized as 57 bytes, to match the point serialization length. Is this a change you're interested in?

Reply:
(I mean, in `to/from_bytes` and not just `from_canonical_bytes`)

Reply:
Yes very interested!